### ISSUE TITLE
Breakpoints are misplaced when running mocha with in-place transpilation 

### ISSUE THREAD

**Issue by blikblum** [microsoft/vscode-js-debug#636]
**Describe the bug**
When running mocha tests with in-place transpilation (@babel/register) the breakpoints are misplaced.

The issue occurs when configuring transpilation with --register option or with a setup script calling @babel/register directly

**To Reproduce**
1. Clone https://github.com/blikblum/vscode-mocha-debug (Node project with mocha configured to run transpilation with @babel/register)
2. Add a breakpoint in `doit.spec.js` at line 5 (call to `doIt(1)`)
3. Run debugger using "Mocha Tests" or "Mocha Tests Setup Script"
4. The debugger will break at the transpiled file at `describe` call (correspond to line 5 in transpiled file)

**Log File**

[vscode-debugadapter-0.json.gz](https://github.com/microsoft/vscode-js-debug/files/4938583/vscode-debugadapter-0.json.gz)

**VS Code Version:**
Version: 1.47.2 (system setup)
Commit: 17299e413d5590b14ab0340ea477cdd86ff13daf
Date: 2020-07-15T18:22:06.216Z (1 day ago)

Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.19041

**Additional context**
Error occurs with both shipped debugger and nightly one

**Comment by connor4312** [comment]
Thanks for the night issue and reproduction, this will be fixed in the [next nightly](https://github.com/microsoft/vscode-js-debug#nightly-extension)

**Comment by connor4312** [comment]
I have this working reliably on Windows, but on OSX it's... hard

Babel seems to like to generate absolute sourceRoots, e.g. `C:/Users/Connor/cool-project`. This is fine on Windows. But on OSX/Linux that turns into `/home/connor/cool-project`. We have no way to tell if the compiler intended that to be an absolute on disk, or if it's generating an absolute path expecting it to be loaded from the root of the URL or 'workspace'. For example, a compiler can generate the source root as `/` and have relative files like `src/foo.ts` and expect that to be relative to the workspace. How can we tell the difference between that -- which we need to re-map to the correct path -- versus paths that are already correct? 

Typing this out I think one decent heuristic is going with the concatenation of the source root and workspace folder/rootPath if the resulting folder exists, and assume the source root is an absolute disk path otherwise. Thoughts @roblourens?

**Comment by blikblum** [comment]
Testing with latest nightly 2020.7.1717 the bug still exists

[vscode-debugadapter-3.json.gz](https://github.com/microsoft/vscode-js-debug/files/4942861/vscode-debugadapter-3.json.gz)

**Comment by connor4312** [comment]
@blikblum you're hitting this, see here: https://github.com/microsoft/vscode-js-debug/issues/563#issuecomment-659592018

**Comment by roblourens** [comment]
I don't think this case was handled in the v1 debugger, but I remember the same struggle. A heuristic like that is not perfect (both can exist 😳) but is probably the best solution
